data_analysis/near-deduplication/minhash_deduplication_alt.py

```diff
@@ -5,697 +5,331 @@
-                    is_false_positive = False
-                    break
-            if is_false_positive:
-                num_false_positives += 1
-                deltas.append(threshold - max_similarity)
-        cluster_false_positive_rates.append(num_false_positives / len(ids))
-
-    logger.info(
-        f"Average false positive rate from {len(clusters)} clusters: {np.mean(cluster_false_positive_rates):.2f}"
-    )
-    logger.info(f"Similarity delta stats from threshold:")
-    logger.info(f"-  Max : {np.max(deltas):0.2f}")
-    logger.info(f"-  Min : {np.min(deltas):0.2f}")
-    logger.info(f"-  Mean: {np.mean(deltas):0.2f}")
-    logger.info(f"-  Std : {np.std(deltas):0.2f}")
-
-
-def find_duplicate_communities(
-    records: Iterable | Dataset,
-    community_detection: bool,
-    output: str,
-    report_false_positive_rate: bool = False,
-    reference_records: Iterable | Dataset | None = None,
-    threshold: float = 0.85,
-    column: str = "content",
-    input_graph: str | None = None,
-    output_graph: str | None = None,
-    verbose: bool = False,
-) -> Set[int]:
-    """
-    Find the duplicate communities from the queried dataset.
+    Compute the optimal `MinHashLSH` parameter that minimizes the weighted sum
+    of probabilities of false positive and false negative, taken from datasketch.
 
     Parameters
     ----------
-    records : Iterable | Dataset
-        The dataset that contains both `__id__` and `__neighbors__`.
-    community_detection : bool
-        Whether to use community detection to find the duplicate communities, or to use the connected components.
-    output : str
-        The output file to save the duplicate communities.
-    report_false_positive_rate : bool
-        Whether to report the false positive rate.
-    reference_records : Iterable | Dataset | None
-        The reference records. It can be an iterable or a Dataset. It is only used when `report_false_positive_rate` is True.
     threshold : float
-        The threshold to use for calculating the false positive rate. It is only used when `report_false_positive_rate` is True.
-    column : str
-        The column to use for calculating the false positive rate. It is only used when `report_false_positive_rate` is True.
-    input_graph : str | None
-        The input graph file to load the graph from.
-    output_graph : str | None
-        The output graph file to save the graph to.
-    verbose : bool
-        Whether to print verbose logs and false positive rate stats.
+        The threshold for similarity.
+    num_perm : int
+        The number of permutations.
+    false_positive_weight : float
+        The weight of false positive.
+    false_negative_weight : float
+        The weight of false negative.
 
     Returns
     -------
-    Set[int]
-        The set of duplicate ids that should be removed, leaving only one id in each community.
+    Tuple[int, int]
+        The optimal `b` and `r` parameters.
```

Comment:
maybe mention what these parameters are instead of their one-letter names?

Reply:
Good point, I can update this. The assumption here, still, is that the target audience knows enough about these standard two parameters used for LSH.